addon/addon/resolvers/classic/index.js

```diff
@@ -130,22 +128,22 @@ function resolveOther(parsedName) {
   }
 }
 
-const Resolver = EmberObject.extend({
-  resolveOther,
-  parseName,
-  pluralizedTypes: null,
-  moduleRegistry: null,
+class Resolver extends EmberObject {
+  resolveOther = resolveOther;
+  parseName = parseName;
```

Comment:
The way this was being done before allowed for consumers to use `this._super()` within `resolveOther` and `parseName` (because the POJO you pass to `.extend()` is augmenting the prototype not the instance), but rewriting to class fields like this means that will no longer work.

Let's either define them as methods (that just call the module functions) like this:

```
class Resolver extends EmberObject {
  resolveOther(parsedName) {
    return resolveOther(parsedName);
  }
}
```

Or just move the function definitions inline into the class body (I'd probably prefer this?). I do not believe they are hoisted for any "good" reason (other than that was how the code was originally authored way back in 2013 :P ).